DataAnnotationsValidator/DataAnnotationsValidator/DataAnnotationsValidator.cs

```diff
@@ -66,10 +64,10 @@ private bool TryValidateObjectRecursive<T>(T obj, List<ValidationResult> results
                         result = false;
                         foreach (var validationResult in nestedResults)
                         {
-                            PropertyInfo property1 = property;
-                            results.Add(new ValidationResult(validationResult.ErrorMessage, validationResult.MemberNames.Select(x => property1.Name + '.' + x)));
+                            results.Add(new ValidationResult(validationResult.ErrorMessage, 
+                                validationResult.MemberNames.Select(member => $"{property.Name}{'.'}{member}")));
                         }
-                    };
+                    }
```

Comment:
i'm onboard with the code changes, but i'd typically prefer a separate PR since this one is about target versions